### DISCUSSION TITLE
Using xonsh as a notebook

### DISCUSSION THREAD

**Discussion by mrsipan** [xonsh/xonsh#5121] [Q&A]
I know that _xonsh_ have been designed to be a shell. However, it is so close to behave as a totally capable notebook that it begs the question of whether someone else has been using it as a seudo-notebook. _Euporie_ works well as a jupyter notebook in the terminal, but it has too much going on. Rather I was thinking of an interactive _xonsh_ session in which I could add "text cells" to comment or explain the flow of my resoning while on the the xonsh shell.

The missing bit personally is that there is not a way to add such a "text cell" interactively. What I am currently doing is to use triple quotes. The text inside the quotes are kept in the xonsh history so, it kind of does the job.

Wonder if someone else has been doing this.

**Comment by anki-code** [comment]
JFYI Euporie notebook that support xonsh kernel via [xontrib-jupyter](https://github.com/xonsh/xontrib-jupyter) can switch cell type - https://euporie.readthedocs.io/en/latest/apps/notebook.html#changing-a-cell-s-type

**Comment by majidaldo** [comment]
i think the missing link is just something that intercepts the output and renders html if its html.

**Reply by majidaldo** [reply]
> i think the missing link is just something that intercepts the output and renders html if its html.

@mrsipan , https://quarto.org/ is in this direction. even if it's less 'shell'-y i'm finding really useful and better than jupyter.
